pkg/postgresql/postgresql.go

```diff
@@ -157,24 +164,47 @@ out:
 	return nil
 }
 
+// Start starts the PostgreSQL server
 func (p *Manager) Start() error {
-	log.Infof("Starting database")
+	log.Info("Starting database")
 	if err := p.WriteConf(); err != nil {
-		return fmt.Errorf("error writing conf file: %v", err)
+		return trace.Wrap(err, "error writing conf file")
 	}
 	name := filepath.Join(p.pgBinPath, "pg_ctl")
-	cmd := exec.Command(name, "start", "-w", "-D", p.dataDir)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	// TODO(sgotti) attaching a pipe to sdtout/stderr makes the postgres
-	// process executed by pg_ctl inheriting it's file descriptors. So
-	// cmd.Wait() will block and waiting on them to be closed (will happend
-	// only when postgres is stopped). So this functions will never return.
-	// To avoid this no output is captured. If needed there's the need to
-	// find a way to get the output whitout blocking.
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("error: %v", err)
+	cmd := exec.Command(name, "start", "-w", "-t",
+		strconv.FormatInt(int64(startTimeout/time.Second), 10), "-D", p.dataDir)
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return trace.Wrap(err)
 	}
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	log.Infof("Executing command: %s", cmd.String())
+	if err = cmd.Start(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	stdoutScanner := bufio.NewScanner(stdoutPipe)
```

Comment:
You could additionally ensure that these goroutines finish before you leave here if you implemented them like this:
```go
wg sync.WaitGroup
wg.Add(2)
go logOutput(stdoutPipe, wg, "stdout")
go logOutput(stdoutPipe, wg, "stderr")
if err = cmd.Wait(); err != nil {
	return trace.Wrap(err)
}
wg.Wait()
```
```go
func logOutput(r io.Reader, wg sync.WaitGroup, pipeName string) {
	scanner := bufio.NewScanner(r)
	for scanner.Scan() {
		log.Infof("pg_ctl command %v: %s", pipeName, scanner.Text())
	}
	wg.Done()
}
```